crates/notion-core/src/tool/npm.rs

```diff
@@ -22,13 +22,21 @@ impl Tool for Npm {
             throw!(ErrorDetails::NoGlobalInstalls);
         }
 
-        if let Some(ref platform) = session.current_platform()? {
+        // if we're in a pinned project, use npm from that platform
+        if let Some(ref platform) = session.project_platform()? {
             let image = platform.checkout(session)?;
             Ok(Self::from_components(
                 OsStr::new("npm"),
                 args,
                 &image.path()?,
             ))
+        } else if let Some(user_tool) = session.get_user_tool(OsStr::new("npm"))? {
+            // npm is installed as a user tool
+            return Ok(Self::from_components(
```

Comment:
I actually removed this `else if` for now (from @charlespierce's comments above), because it is not possible to install npm as a user tool yet, and this is not exactly the right way to execute it